docs/CMP-V1-TO-V2.md

```diff
@@ -0,0 +1,312 @@
+                }
+            }, data.listenerId);
+            googletag.cmd.push(function() {
+                googletag.display('my-div-id');
+            });
+        }
+    });
+}
+```
+Note that the `removeEventListener` command, used in the v2 code, is used to ensure the `googletag.cmd.push` method is invoked only once.
+Without running the `removeEventListener` command, the `googletag.cmd.push` method would be invoked every time a user re-confirms their choices.
+
+### The `getNonIABVendorConsents` command
+Quantcast v1 implements the command `getNonIABVendorConsents`.
+It invokes the callback with an object containing the consent given to the non-IAB vendors.
+
+Quantcast v2 also implements the command `getNonIABVendorConsents`.
+
+The object given to the v2 callback differs from the object given to the v1 callback.
+In v1 the object has a property named _non**IAB**VendorConsents_, while in v2 the object has a property named _non**Iab**VendorConsents_.
+
+The following example shows how to refactor the `getNonIABVendorConsents` command:
+
+Before:
+```js
+function handleNonIabVendorConsents(nonIabVendorConsents) {
+    const nonIabVendorId = 36;
+    if (nonIabVendorConsents[nonIabVendorId] === true) {
+        console.log('Non-IAB vendor has consent');
+    } else if (nonIabVendorConsents[nonIabVendorId] === false) {
+        console.log('Non-IAB vendor does not have consent');
+    } else {
+        console.log('Unknown if non-IAB vendor has consent');    
+    }
+}
+
+window.__cmp('getNonIABVendorConsents', null, function(consent, success) {
+    if (success) {
+        handleNonIabVendorConsents(consent.nonIABVendorConsents)        
+    }
+});
+```
+After:
+```js
+function handleNonIabVendorConsents(nonIabVendorConsents) {
+    const nonIabVendorId = 36;
+    if (nonIabVendorConsents[nonIabVendorId] === true) {
+        console.log('Non-IAB vendor has consent');
+    } else if (nonIabVendorConsents[nonIabVendorId] === false) {
+        console.log('Non-IAB vendor does not have consent');
+    } else {
+        console.log('Unknown if non-IAB vendor has consent');    
+    }
+}
+
+if (!window.__cmp && !window.__tcfapi) {
+    console.error('both __cmp and __tcfapi are undefined');
+}
+if (window.__cmp) {
+    window.__cmp('getNonIABVendorConsents', null, function(consent, success) {
+        if (!success) {
+            console.error('getNonIABVendorConsents was not successful');
+            return;
+        }
+        handleNonIabVendorConsents(consent.nonIABVendorConsents);
+    });
+}
+if (window.__tcfapi) {
+    window.__tcfapi('addEventListener', 2, function(tcData, addSuccess) {
+        if (addSuccess && tcData.eventStatus === 'useractioncomplete') {
+            window.__tcfapi('getNonIABVendorConsents', 2, function(consent, nonIabVendorSuccess) {
+                window.__tcfapi('removeEventListener', 2, function(removeSuccess) {
```

Comment:
Shouldn't you only remove the eventListener if the `getNonIABVendorConsents` was succesfull?